lua_modules/dht22/dht22.lua

```diff
@@ -0,0 +1,101 @@
+-- ***************************************************************************
+-- DHT22 module for ESP8266 with nodeMCU
+--
+-- Written by Javier Yanez 
+-- but based on a script of Pigs Fly from ESP8266.com forum
+--
+-- MIT license, http://opensource.org/licenses/MIT
+-- ***************************************************************************
+
+local moduleName = ...
+local M = {}
+_G[moduleName] = M
+
+local humidity
+local temperature
+local checksum
+local checksumTest
+
+function M.read(pin)
+  humidity = 0
+  temperature = 0
+  checksum = 0
+
+  -- Use Markus Gritsch trick to speed up read/write on GPIO
+  gpio_read = gpio.read
```

Comment:
I made a test:

```lua
function test()
  pin = 4
  gpio_read = gpio.read
  local gpio_read_local = gpio.read

  c = 0
  time1 = tmr.now() 
  while (gpio.read(pin) == 1 and c < 100) do c = c + 1 end
  time2 = tmr.now()
  print("Time: "..(time2 - time1))
  print(c)

  c = 0
  time1 = tmr.now() 
  while (gpio_read(pin) == 1 and c < 100) do c = c + 1 end
  time2 = tmr.now()
  print("Time: "..(time2 - time1))
  print(c)
  
  c = 0
  time1 = tmr.now() 
  while (gpio_read_local(pin) == 1 and c < 100) do c = c + 1 end
  time2 = tmr.now()
  print("Time: "..(time2 - time1))
  print(c)
end

test()
```

The result has been that:

>Time: 6170
100
Time: 2191
100
Time: 2048
100

As you you said, it's a little faster in a local variable. If c is local then the speed increase is great.

As the loop speed is critical, I will check if you need some adjustment before making the change.